lib/ansible/modules/cloud/amazon/ec2_eni_facts.py

```diff
@@ -66,122 +256,65 @@
-                      'subnet_id': interface.subnet_id,
-                      'vpc_id': interface.vpc_id,
-                      'description': interface.description,
-                      'owner_id': interface.owner_id,
-                      'status': interface.status,
-                      'mac_address': interface.mac_address,
-                      'private_ip_address': interface.private_ip_address,
-                      'source_dest_check': interface.source_dest_check,
-                      'groups': dict((group.id, group.name) for group in interface.groups),
-                      'private_ip_addresses': private_addresses
-                      }
-
-    if hasattr(interface, 'publicDnsName'):
-        interface_info['association'] = {'public_ip_address': interface.publicIp,
-                                         'public_dns_name': interface.publicDnsName,
-                                         'ip_owner_id': interface.ipOwnerId
-                                         }
-
-    if interface.attachment is not None:
-        interface_info['attachment'] = {'attachment_id': interface.attachment.id,
-                                        'instance_id': interface.attachment.instance_id,
-                                        'device_index': interface.attachment.device_index,
-                                        'status': interface.attachment.status,
-                                        'attach_time': interface.attachment.attach_time,
-                                        'delete_on_termination': interface.attachment.delete_on_termination,
-                                        }
-
-    return interface_info
-
-
-def list_eni(connection, module):
-
-    filters = module.params.get("filters")
-    interface_dict_array = []
+    network_interface_ids = module.params.get("network_interface_ids")
+    filters = ansible_dict_to_boto3_filter_list(module.params.get("filters"))
 
     try:
-        all_eni = connection.get_all_network_interfaces(filters=filters)
-    except BotoServerError as e:
-        module.fail_json(msg=e.message)
+        network_interfaces = connection.describe_network_interfaces(NetworkInterfaceIds=network_interface_ids, Filters=filters)['NetworkInterfaces']
+    except ClientError as e:
+        module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
 
-    for interface in all_eni:
-        interface_dict_array.append(get_eni_info(interface))
+    # Turn the boto3 result in to ansible_friendly_snaked_names
+    snaked_network_interfaces = []
+    for network_interface in network_interfaces:
+        # Copy the tags because we don't want to snake these
+        if 'TagSet' in network_interface:
+            tag_set_copy = copy.copy(network_interface['TagSet'])
+        snaked_network_interfaces.append(camel_dict_to_snake_dict(network_interface))
+        snaked_network_interfaces['tags'] = boto3_tag_list_to_ansible_dict(tag_set_copy)
```

Comment:
Ah, it's nice that the syntax checker caught this bug.

This should be `network_interface['tags'] = boto3_tag_list_to_ansible_dict(tag_set_copy)`